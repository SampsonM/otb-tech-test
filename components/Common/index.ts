export { default as PageWrapper } from './PageWrapper/PageWrapper'
export { H3 as H3 } from './Typography/Typography'
export { P as P } from './Typography/Typography'
export { Strong as Strong } from './Typography/Typography'
export { PrimaryButton as PrimaryButton } from './Button/Button'
