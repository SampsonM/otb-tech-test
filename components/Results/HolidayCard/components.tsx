import styled from 'styled-components'
import Image from 'next/image'
import { colours, deviceWidthQuery } from '../../../constants/styles'

export const HolidayCardContainer = styled.div`
	display: flex;
	flex-direction: column;
	background-color: ${colours.primaryWhite};
	margin-bottom: 20px;
`

export const ImageAndDetails = styled.div`
	display: flex;
	flex-direction: column;
	height: 100%;

	@media ${deviceWidthQuery.tablet} {
		flex-direction: row;
		height: 410px;
	}
	
	@media ${deviceWidthQuery.laptop} {
		height: 340px;
	}
`

export const StyledImageWrapper = styled.div`
	position: relative;
	max-width: 100%;
	height: 250px;
	
	@media ${deviceWidthQuery.tablet} {
		height: auto;
		width: 65%;
	}
`

export const StyledImage = styled(Image)`
	width: 100%;
	height: 100%;
	object-fit: cover;
`
