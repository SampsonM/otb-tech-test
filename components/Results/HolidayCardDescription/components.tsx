import styled from "styled-components";
import { colours, deviceWidthQuery } from '../../../constants/styles'

export const DescriptionContainer = styled.div`
	position: relative;
`

export const DescriptionWrapper = styled.div`
	width: 100%;
	height: auto;
	padding: 10px;
	
	@media ${deviceWidthQuery.tablet} {
		padding: 15px;
	}
`

export const ReadMoreButton = styled.button`
	display: flex;
	align-items: center;
	justify-content: center;
	border: 0;
	background-color: ${colours.primaryWhite};
	height: 30px;
	text-align: left;
	width: 100%;

	&:hover {
		background-color: ${colours.secondaryGrey};
	}

	@media ${deviceWidthQuery.tablet} {
		width: auto;
		min-width: 253px;
		margin: 0 auto;
    position: absolute;
    top: -36px;
    height: auto;
	}
`
