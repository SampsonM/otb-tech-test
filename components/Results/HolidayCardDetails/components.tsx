import { H3, P, Strong } from "@/components/Common/Typography/Typography"
import styled from "styled-components"
import { deviceWidthQuery } from "../../../constants/styles"

export const DetailsContainer = styled.div`
	padding: 10px;
	
	@media ${deviceWidthQuery.tablet} {
		padding: 15px;
		width: 35%;
	}
`

export const LocationRatingContainer = styled.div`
	display: flex;
	flex-direction: row;
	
	@media ${deviceWidthQuery.tablet} {
		flex-direction: column;
	}
`

export const HotelLocation = styled(P)`
	color: #747373;
	font-style: italic;
	margin-right: 10px;

	@media ${deviceWidthQuery.tablet} {
		margin-right: 0;
	}
`

export const TotalPrice = styled(Strong)`
	font-size: 25px;
`
