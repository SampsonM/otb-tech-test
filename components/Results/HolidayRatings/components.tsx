import styled from 'styled-components'

export const RatingsWrapper = styled.div`
	display: flex;
	align-items: center;
`
