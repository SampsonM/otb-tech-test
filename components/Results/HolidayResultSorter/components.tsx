import styled from 'styled-components'
import { colours, deviceWidthQuery } from '@/constants/styles'

type SortButtonProps = {
	active: boolean;
}

export const SortButton = styled.button<SortButtonProps>`
	display: flex;
	align-items: center;
	justify-content: space-between;
	width: 100%;
	height: 40px;
	background-color: ${props => props.active ? colours.primaryBlue : colours.primaryWhite};
	color: ${props => props.active ? colours.primaryWhite : colours.primaryBlue};
	border: 0;
	margin-bottom: 1px;
	border: 2px solid transparent;

	&:hover {
		border: 2px solid ${colours.primaryBlue};
		background-color: ${colours.secondaryGrey};
		color: ${colours.primaryBlue};
	}
`

export const SorterContainer = styled.div`
	margin-bottom: 5px;

	// Ideally this would be 100% width and be controlled by the consumer component
	// so the sorter buttons could slot into anywhere they were needed
	@media ${deviceWidthQuery.tablet} {
		width: 25%;
	}
`
