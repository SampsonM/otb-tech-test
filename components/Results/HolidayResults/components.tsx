import { deviceWidthQuery } from "@/constants/styles"
import styled from "styled-components"

export const HolidayResultsWrapper = styled.div`
	width: 100%;

	@media ${deviceWidthQuery.tablet} {
		width: 60%;
	}
`