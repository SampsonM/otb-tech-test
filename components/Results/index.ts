export { default as HolidayCard } from './HolidayCard/HolidayCard'
export { default as HolidayCardDescription } from './HolidayCardDescription/HolidayCardDescription'
export { default as HolidayCardDetails } from './HolidayCardDetails/HolidayCardDetails'
export { default as HolidayRatings } from './HolidayRatings/HolidayRatings'
export { default as HolidayResults } from './HolidayResults/HolidayResults'
export { default as HolidayResultSorter } from './HolidayResultSorter/HolidayResultSorter'
export { default as HolidaySortAndResultsContainer } from './HolidaySortAndResultsContainer/HolidaySortAndResultsContainer'
