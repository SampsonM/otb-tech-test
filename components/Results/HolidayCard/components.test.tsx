import React from "react"
import { render } from "@testing-library/react"
import { HolidayCardContainer, StyledImage } from './components'

describe("Elements", () => {
	test("HolidayCardContainer fragment matches the snapshot", () => {
		const { asFragment } = render(<HolidayCardContainer />)
		expect(asFragment()).toMatchSnapshot()
	})

	test("StyledImage fragment matches the snapshot", () => {
		const { asFragment } = render(
			<StyledImage
				src="/test-url.png"
				alt="Picture of a hotelName"
				height={100}
				width={100}
			/>)
		expect(asFragment()).toMatchSnapshot()
	})
})
