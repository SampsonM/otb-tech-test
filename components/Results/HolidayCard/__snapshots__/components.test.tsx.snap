// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements HolidayCardContainer fragment matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="components__HolidayCardContainer-otb__sc-40465409-0 eENjQW"
  />
</DocumentFragment>
`;

exports[`Elements StyledImage fragment matches the snapshot 1`] = `
<DocumentFragment>
  <img
    alt="Picture of a hotelName"
    class="components__StyledImage-otb__sc-40465409-3 fxKFaB"
    data-nimg="1"
    decoding="async"
    height="100"
    loading="lazy"
    src="/_next/image?url=%2Ftest-url.png&w=256&q=75"
    srcset="/_next/image?url=%2Ftest-url.png&w=128&q=75 1x, /_next/image?url=%2Ftest-url.png&w=256&q=75 2x"
    style="color: transparent;"
    width="100"
  />
</DocumentFragment>
`;
